test/fail_compilation/fail18228.d

```diff
@@ -0,0 +1,13 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail18228.d(11): Error: `this` cannot be used as a parameter type, use `typeof(this)` instead
+fail_compilation/fail18228.d(12): Error: `this` cannot be used as a parameter type, use `typeof(this)` instead
```

Comment:
This is a [comma splice](https://owl.english.purdue.edu/engagement/2/1/34/). Please change the comma to a semi-colon or period.

Reply:
@MetaLang do you have the rights to make changes to PRs?

Reply:
Done. Thanks!